src/rgw/rgw_op.cc

```diff
@@ -968,10 +966,8 @@ int handle_cloudtier_obj(req_state* s, const DoutPrefixProvider *dpp, rgw::sal::
       if (restore_op) {
         op_ret = -ERR_INVALID_OBJECT_STATE;
         s->err.message = "only cloud tier object can be restored";
-        return op_ret;
-      } else { //ignore for read-through
-        return 0;
       }
+      return op_ret;
```

Comment:
Please retain the message "ignore for read-through" to avoid any confusion later.